pkg/resources/task.go

```diff
@@ -204,51 +225,43 @@ func ImportTask(ctx context.Context, d *schema.ResourceData, meta any) ([]*schem
 	return []*schema.ResourceData{d}, nil
 }
 
-func CreateTask(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+func CreateTask(ctx context.Context, d *schema.ResourceData, meta any) (diags diag.Diagnostics) {
 	client := meta.(*provider.Context).Client
 
 	databaseName := d.Get("database").(string)
 	schemaName := d.Get("schema").(string)
 	name := d.Get("name").(string)
 	id := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
-	req := sdk.NewCreateTaskRequest(id, d.Get("sql_statement").(string))
 
+	req := sdk.NewCreateTaskRequest(id, d.Get("sql_statement").(string))
 	tasksToResume := make([]sdk.SchemaObjectIdentifier, 0)
 
-	if v, ok := d.GetOk("warehouse"); ok {
-		warehouseId, err := sdk.ParseAccountObjectIdentifier(v.(string))
-		if err != nil {
-			return diag.FromErr(err)
-		}
-		req.WithWarehouse(*sdk.NewCreateTaskWarehouseRequest().WithWarehouse(warehouseId))
-	}
-
-	if v, ok := d.GetOk("schedule"); ok {
-		req.WithSchedule(v.(string))
-	}
-
-	if v, ok := d.GetOk("config"); ok {
-		req.WithConfig(v.(string))
-	}
-
-	if v := d.Get("allow_overlapping_execution").(string); v != BooleanDefault {
-		parsedBool, err := booleanStringToBool(v)
-		if err != nil {
-			return diag.FromErr(err)
-		}
-		req.WithAllowOverlappingExecution(parsedBool)
-	}
-
-	if v, ok := d.GetOk("error_integration"); ok {
-		notificationIntegrationId, err := sdk.ParseAccountObjectIdentifier(v.(string))
-		if err != nil {
-			return diag.FromErr(err)
-		}
-		req.WithErrorIntegration(notificationIntegrationId)
-	}
-
-	if v, ok := d.GetOk("comment"); ok {
-		req.WithComment(v.(string))
+	if errs := errors.Join(
+		attributeMappedValueCreate(d, "warehouse", &req.Warehouse, func(v any) (*sdk.CreateTaskWarehouseRequest, error) {
+			warehouseId, err := sdk.ParseAccountObjectIdentifier(v.(string))
+			if err != nil {
+				return nil, err
+			}
+			return sdk.NewCreateTaskWarehouseRequest().WithWarehouse(warehouseId), nil
+		}),
+		attributeMappedValueCreate(d, "schedule", &req.Schedule, func(v any) (*string, error) {
+			if len(v.([]any)) > 0 {
+				if minutes, ok := d.GetOk("schedule.0.minutes"); ok {
+					return sdk.String(fmt.Sprintf("%d MINUTE", minutes)), nil
+				}
+				if cron, ok := d.GetOk("schedule.0.using_cron"); ok {
+					return sdk.String(fmt.Sprintf("USING CRON %s", cron)), nil
+				}
+			}
+			return nil, nil
```

Comment:
if we get here, we either had len == 0 or both minutes and using cron were not OK. The second case should not have nil err

Reply:
Actually, I had to modify it again because schema validation doesn't work (checked after the test), so I removed the validation from "minutes" and "using_cron" and moved it here.

Reply:
Yeah, no problem, added